services/galley/src/Galley/API/Query.hs

```diff
@@ -117,6 +118,19 @@ getConversations user mids mstart msize = do
       | Data.isConvDeleted c = Data.deleteConversation (Data.convId c) >> pure False
       | otherwise = pure True
 
+iterateConversations :: UserId -> Range 1 500 Int32 -> ([Public.Conversation] -> Galley ()) -> Galley ()
+iterateConversations uid pageSize handleConvs = go Nothing
+  where
+    go mbConv = do
+      convResult <- getConversations uid Nothing mbConv (Just pageSize)
+      handleConvs (convList convResult)
+      case convList convResult of
+        (conv : rest) ->
+          if convHasMore convResult
+            then go (Just (maximum (cnvId <$> (conv : rest))))
+            else pure ()
+        _ -> pure ()
```

Comment:
```suggestion
      if convHasMore convResult && nonEmpty (convList convResult)
        then go (Just (maximum (cnvId <$> convList convResult)))
        else pure ()
```

But that's still too complicated, no?  Either the cnvIds are ordered, then `last` instead of `maximum` would also work, or they are not, then `maximum` won't work either.

Reply:
Why wouldn't `maximum` always work?

Reply:
If the output is not ordered, there will probably be elements in the next page that are smaller than the maximum of this page.  Anyway in that case it wouldn't even make sense to say "give me the next page starting after X".

Do we maybe have pagination code somewhere in wire-server from which we could steal?